src/program.js

```diff
@@ -258,55 +217,49 @@ export default class Program {
 
   // Alternate constructor
   // Create a program from vertex and fragment shader node ids
-  static async fromShaderIds(...args) {
-    const opt = Program._getOptions({}, ...args);
-    const vertexShader = document.getElementById(opt.vs).innerHTML;
-    const fragmentShader = document.getElementById(opt.fs).innerHTML;
-    return new Program(opt.app, vertexShader, fragmentShader);
+  static fromHTMLTemplates(app, vs, fs) {
+    const vertexShader = document.getElementById(vs).innerHTML;
+    const fragmentShader = document.getElementById(fs).innerHTML;
+    return new Program(app, vertexShader, fragmentShader);
   }
 
   // Alternate constructor
   // Create a program from vs and fs sources
-  static async fromShaderSources(...args) {
-    var opt = Program._getOptions({}, ...args);
-    return new Program(opt.app, opt.vs, opt.fs);
+  static fromShaderSources(app, vs, fs) {
```

Comment:
I think we can drop this constructor and just have applications use `new Program`.